packages/use-contractkit/src/utils/useDappVersion.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+
+import { ChainId } from '..';
+import type { Dapp } from '../types';
+
+export type WalletConnectRegistryDapp = {
+  app: {
+    browser: string | null;
+    ios: string | null;
+    android: string | null;
+    mac: string | null;
+    windows: string | null;
+  };
+  description: string;
+  chains: string[];
+  desktop: {
+    native: string;
+    universal: string;
+  };
+  mobile: {
+    native: string;
+    universal: string;
+  };
+  homepage: string;
+  id: string;
+  metadata: {
+    shortName: string;
+    colors: Record<string, string>;
+  };
+  name: string;
+  versions: string[];
+};
+export type WalletConnectRegistry = { [x: string]: WalletConnectRegistryDapp };
+
+const WALLETCONNECT_REGISTRY_URL =
+  'https://raw.githubusercontent.com/WalletConnect/walletconnect-registry/production/public/data/dapps.json';
+
+export function useDappVersion(dapp: Dapp): number | null {
```

Comment:
I don't think the "dapp" argument is used at all.

Reply:
I hope this isn't a huge pain, but maybe we should name this stuff after wallets instead of dapps. So maybe
`useWalletVersion`